app/src/main/java/com/conupods/OutdoorMaps/View/SearchView/AbstractCampusLocationAdapter.java

```diff
@@ -0,0 +1,109 @@
+package com.conupods.OutdoorMaps.View.SearchView;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Filter;
+import android.widget.Filterable;
+import android.widget.TextView;
+
+import com.conupods.OutdoorMaps.Models.Buildings.AbstractCampusLocation;
+import com.conupods.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+public class AbstractCampusLocationAdapter extends RecyclerView.Adapter<AbstractCampusLocationAdapter.MyViewHolder> implements Filterable {
+
+    private Context mCurrentContext;
+    private List<AbstractCampusLocation> mCampusLocations;
+    private List<AbstractCampusLocation> mFilteredCampusLocationsList;
+    private CampusLocationsAdapterListener mCampusLocationsAdapterListener;
+
+
+    public class MyViewHolder extends RecyclerView.ViewHolder {
+        public TextView mIdentifierText, mPhysicalParentNameText;
+
+        public MyViewHolder(View view) {
+            super(view);
+            mIdentifierText = (TextView) view.findViewById(R.id.AbstractLocationName);
+            mPhysicalParentNameText = (TextView) view.findViewById(R.id.BuildingName);
+
+            view.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    mCampusLocationsAdapterListener.onCampusLocationSelected(mFilteredCampusLocationsList.get(getAdapterPosition()));
+                }
+            });
+
+        }
+    }
+
+
+    public AbstractCampusLocationAdapter(Context context, List<AbstractCampusLocation> listOfCampusLocations, CampusLocationsAdapterListener campusLocationsAdapterListener){
+        mCurrentContext = context;
```

Comment:
The internet says 
> It is usually in your best interest to just pass the current context at the moment it is needed. Storing it in a member variable will likely lead to leaked memory, and start causing issues as you build out more Activities and Services in your app.

if you need context in any of your methods, just call `App.getContext()`, I don't think the context changes across the app, it is always the same.

Reply:
removed the Context object